packages/core/cache/src/FSCache.js

```diff
@@ -151,6 +151,21 @@ export class FSCache implements Cache {
     await Promise.all(writePromises);
   }
 
+  async deleteLargeBlob(key: string): Promise<void> {
+    const deletePromises: Promise<void>[] = [];
+
+    let i = 0;
+    let filePath = this.#getFilePath(key, i);
+
+    while (await this.fs.exists(filePath)) {
```

Comment:
/nit `fs.exists` (as opposed to `existsSync`) prior to performing a file operation is "[deprecated](https://nodejs.org/api/fs.html#fsexistspath-callback)" as the existence could change between the two operations.

It's an edge case, but you should probably use `existsSync` here. I believe `rimraf` itself will never "fail" if the file doesn't exist.

Reply:
Good call, I'll fix it up in my next PR